### PR TITLE
[JN-1053] Withdrawal journey for proxies

### PR DESCRIPTION
**Pull request by connorlbark**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

Updates governed user withdrawal to withdrawal the proxy at the same time.

Right now it isn't possible to enroll yourself and a governed user as well. I'll need to add that in an upcoming PR. As such, testing the edge case is impossible.

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

- Observe unit tests for edge case
- Enroll yourself as a proxy
- Withdraw the governed user
- Observe that your proxy account is also withdrawn

### CODE REVIEW

File: core/src/main/java/bio/terra/pearl/core/service/participant/WithdrawnEnrolleeService.java [hunk 1]
```diff
@@ -1,106 +1,137 @@
...
-      return withdrawnEnrollee;
-    } catch (JsonProcessingException e) {
-      throw new InternalServerException("Error serializing enrollee or user data", e);
+    public boolean isWithdrawn(String shortcode) {
+        return dao.isWithdrawn(shortcode);
     }
-  }
-
-  @Transactional
-  public List<WithdrawnEnrollee> withdrawFromPortal(PortalParticipantUser ppUser) throws JsonProcessingException {
```
Comment (connorlbark) [inline core/src/main/java/bio/terra/pearl/core/service/participant/WithdrawnEnrolleeService.java:86]: Were there any plans for this method (+ `withdrawFromJuniper` below)? They were unused, so I went ahead and removed them.

**Review by devonbush — APPROVED** [review]
Looks good, thanks for the conversation to confirm the functionality

**Review by MatthewBemis — APPROVED** [review]
